### PR TITLE
Change entity outline depending on colour and mode. #13211

### PR DESCRIPTION
**Pull request by a19clasv**
Added class 'text' to IE and UML parts that have a stroke color that can possibly be toggled and made the borders change as soon as dark mode is toggled in the diagram.

### CODE REVIEW

**Review by a21aldsa — COMMENTED** [review]
Sure, the issue is resolved. But the issue itself is nonsensical, why would anyone want white borders on a white element? This defeats the purpose of borders, I have attached screenshots that show what I mean. @c21adano @b21oscpe 

image

image

**Comment by c21adano** [comment]
The outline should be white in darkmode if the entity itself has a black colour selected. Its seems i described the first part of the issue poorly and i apologize. If the entity has a white colour it should have the normal outline.

**Comment by a19clasv** [comment]
I fixed it

File: DuggaSys/diagram.js [hunk 1]
```diff
@@ -11410,21 +11411,23 @@ function toggleBorderOfElements() {
...
                 //assign their current stroke color to a variable.
-                let strokeColor = text.getAttribute('stroke');
-                //if the element has a stroke which has the color #383737: set it to white.
-                //this is because we dont want to affect the strokes that are null or other colors.
-                if (strokeColor == '#383737') {
+              let strokeColor = text.getAttribute('stroke');
+                let fillColor = text.getAttribute('fill');
+                //if the element has a stroke which has the color #383737 and its fill isn't white: set it to white.
+                //this is because we dont want to affect the strokes that are null or other colors and have a contrasting border.
+                if (strokeColor == '#383737' && fillColor != '#ffffff') {
```
Comment (a21aldsa) [inline DuggaSys/diagram.js:11418]: if its grey, its automatically not white. that means this check is extraneous, or am I misunderstanding something?

**Review by a21aldsa — CHANGES_REQUESTED** [review]
While yes, the border becomes white when the color of the element is set to anything other than white and the theme is dark; you forgot to account for other element colors such as green or pink. Your changes now makes it very difficult to see the border on these elements.

image

This will lead to some users having a hard time differentiating between, for example, the different fields in a UML diagram.

Albeit, this might very well be a separate issue in and of itself. If this is the case I am ok with calling this issue resolved.

**Comment by a21aldsa** [comment]
Also, while it is very easy to see the inside border color of a white element on darkmode now, it is very hard to see the outside border. I believe this should also be resolved or taken up in another issue.

**Comment by c21adano** [comment]
> Also, while it is very easy to see the inside border color of a white element on darkmode now, it is very hard to see the outside border. I believe this should also be resolved or taken up in another issue.
> 
> image

This could be an issue in the future but doesn't need an immediate fix.

**Comment by a19clasv** [comment]
Just give me another issue or expand this issue, stating which border colours should be according to which element colours and which mode. Otherwise this issue is resolved.

**Comment by b21oscpe** [comment]
Discussion provides future issues, the original issue is resolved

**Pull request merged**